src/Service/SaferPayObtainPaymentMethods.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ *NOTICE OF LICENSE
+ *
+ *This source file is subject to the Open Software License (OSL 3.0)
+ *that is bundled with this package in the file LICENSE.txt.
+ *It is also available through the world-wide-web at this URL:
+ *http://opensource.org/licenses/osl-3.0.php
+ *If you did not receive a copy of the license and are unable to
+ *obtain it through the world-wide-web, please send an email
+ *to license@prestashop.com so we can send you a copy immediately.
+ *
+ *DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ *versions in the future. If you wish to customize PrestaShop for your
+ *needs please refer to http://www.prestashop.com for more information.
+ *
+ *@author INVERTUS UAB www.invertus.eu  <support@invertus.eu>
+ *@copyright SIX Payment Services
+ *@license   SIX Payment Services
+ */
+
+namespace Invertus\SaferPay\Service;
+
+use Invertus\SaferPay\Api\Request\ObtainPaymentMethodsService;
+use Invertus\SaferPay\Service\Request\ObtainPaymentMethodsObjectCreator;
+
+class SaferPayObtainPaymentMethods
+{
+    private $obtainPaymentMethodsService;
+    private $obtainPaymentMethodsObjectCreator;
+
+    public function __construct(
+        ObtainPaymentMethodsService $obtainPaymentMethodsService,
+        ObtainPaymentMethodsObjectCreator $obtainPaymentMethodsObjectCreator
+    ) {
+        $this->obtainPaymentMethodsService = $obtainPaymentMethodsService;
+        $this->obtainPaymentMethodsObjectCreator = $obtainPaymentMethodsObjectCreator;
+    }
+
+    public function obtainPaymentMethods()
+    {
+        $paymentMethods = [];
+
+        try {
+            $paymentMethodsObject = $this->obtainPaymentMethodsService->getPaymentMethods(
+                $this->obtainPaymentMethodsObjectCreator->create()
+            );
+        } catch (Exception $e) {
+            throw new SaferPayApiException('Initialize API failed', SaferPayApiException::INITIALIZE);
+        }
+
+        if (!empty($paymentMethodsObject->PaymentMethods)) {
+            foreach ($paymentMethodsObject->PaymentMethods as $paymentMethodObject) {
+                $paymentMethods[] = [
+                    'paymentMethod' => $paymentMethodObject->PaymentMethod,
+                    'logoUrl' => $paymentMethodObject->LogoUrl,
+                ];
+            }
+        }
+
+        if (!empty($paymentMethodsObject->Wallets)) {
+            foreach ($paymentMethodsObject->Wallets as $wallet) {
+                $paymentMethods[] = [
+                    'paymentMethod' => $wallet->WalletName,
+                    'logoUrl' => $wallet->LogoUrl,
```

Comment:
i see that you can get logo url here but you never use it? maybe we should start using logos from here?

Reply:
I use it in saferpayofficial.php  line 164